### ISSUE TITLE
cy.react should retry

### ISSUE THREAD

**Issue by bahmutov** [abhinaba-ghosh/cypress-react-selector#26]
See details in https://github.com/bahmutov/cypress-react-unit-test/issues/379

Built-in Cypress retry-ability https://on.cypress.io/retry-ability

Example of retrying assertions in this file https://github.com/cypress-io/cypress-xpath/blob/master/src/index.js#L116

**Comment by abhinaba-ghosh** [comment]
H @bahmutov,

To accommodate this change, I needed to change the code from 

```js
if (!elements.length) {
        throw new Error(
          `Component not found ${getIdentifierLogs(component, props, state)}`
        );
      }
```

to

```js
if (!elements.length) {
        console.error(
          `Component not found ${getIdentifierLogs(component, props, state)}`
        );
        return null;
      }
```

But, after changing the code, the error message is not getting printed in the debug mode. Is there a way to print the error messages nicely without throwing them?

**Comment by abhinaba-ghosh** [comment]
As per the comment on https://github.com/bahmutov/cypress-react-unit-test/issues/379, there are two parts -

1. Retry for incoming assertions
2. Retry up to nodes identified  ( in case there no incoming assertion)

For point 1, `cypress-xpath` was an amazing reference. For point 2, would you kindly refer to any docs?

**Comment by abhinaba-ghosh** [comment]
fixed in v2.0.1
